development/prow-installer/pkg/cluster/wrapped.go

```diff
@@ -55,17 +56,36 @@ func (caw *APIWrapper) Create(ctx context.Context, clusterConfig Cluster) error
 			NodePools:      nodePools,
 		}}
 
-	createResponse, err := caw.ClusterService.Create(caw.ProjectID, clusterConfig.Location, ccRequest).Context(ctx).Do()
+	_, err := caw.ClusterService.Create(caw.ProjectID, clusterConfig.Location, ccRequest).Context(ctx).Do()
 	if err != nil {
-		return fmt.Errorf("couldn't create cluster: %w", err)
+		return "", fmt.Errorf("couldn't create cluster: %w", err)
+	} else {
+		log.Infof("successfully created cluster %s", clusterConfig.Name)
 	}
-	log.Printf("%#v\n", createResponse)
-	return nil
+
+	// get kubeconfig
+	kubeconfig := "./.kube/" + clusterConfig.Name + "_config"
+
+	for {
+		cmd := exec.Command("gcloud", "container", "clusters", "get-credentials", fmt.Sprintf(clusterConfig.Name),
+			"--project", caw.ProjectID,
+			"--zone", clusterConfig.Location)
+		cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG=%s", kubeconfig))
+
+		// TODO handle gcloud errors specifically
+		if err := cmd.Run(); err != nil {
+			log.Errorf("error getting kubeconfig %v. Retrying in 5 seconds...", err)
+			time.Sleep(time.Second * 5)
+			continue
+		}
+		break
+	}
+	return kubeconfig, nil
 }
 
 // Delete calls the wrapped GCP api to delete a cluster
 func (caw *APIWrapper) Delete(ctx context.Context, name string, zoneId string) error {
-	deleteResponse, err := caw.ClusterService.Delete(caw.ProjectID, caw.ZoneID, name).Context(ctx).Do()
+	deleteResponse, err := caw.ClusterService.Delete(caw.ProjectID, zoneId, name).Context(ctx).Do()
```

Comment:
For consistency, above you named it location is it location here as well or indeed zone?

Reply:
In Google API code the function header is defined as:
```go
func (r *ProjectsZonesClustersService) Delete(projectId string, zone string, clusterId string) *ProjectsZonesClustersDeleteCall
```
I left the naming for consistency with the API function. I'm not quite sure if this would apply also for regions.